common/persistence/nosql/nosqlplugin/cassandra/workflow_test.go

```diff
@@ -670,6 +670,252 @@ func TestDeleteWorkflowExecution(t *testing.T) {
 	}
 }
 
+func TestSelectAllCurrentWorkflows(t *testing.T) {
+	tests := []struct {
+		name           string
+		shardID        int
+		pageToken      []byte
+		pageSize       int
+		iter           *fakeIter
+		wantExecutions []*persistence.CurrentWorkflowExecution
+		wantErr        bool
+	}{
+		{
+			name:      "nil iter returned",
+			shardID:   1,
+			pageToken: []byte("test-page-token"),
+			pageSize:  10,
+			wantErr:   true,
+		},
+		{
+			name:      "run_id is not permanentRunID so excluded from result",
+			shardID:   1,
+			pageToken: []byte("test-page-token"),
+			pageSize:  10,
+			iter: &fakeIter{
+				mapScanInputs: []map[string]interface{}{
+					{
+						"run_id": &fakeUUID{uuid: "17C305FA-79BB-479E-8AC7-360E956AC01A"},
+					},
+				},
+				pageState: []byte("test-page-token-2"),
+			},
+			wantExecutions: nil,
+		},
+		{
+			name:      "multiple executions",
+			shardID:   1,
+			pageToken: []byte("test-page-token"),
+			pageSize:  10,
+			iter: &fakeIter{
+				mapScanInputs: []map[string]interface{}{
+					{
+						"run_id":         &fakeUUID{uuid: permanentRunID},
+						"domain_id":      &fakeUUID{uuid: "domain1"},
+						"current_run_id": &fakeUUID{uuid: "runid1"},
+						"workflow_id":    "wfid1",
+						"workflow_state": 1,
+					},
+					{
+						"run_id":         &fakeUUID{uuid: permanentRunID},
+						"domain_id":      &fakeUUID{uuid: "domain1"},
+						"current_run_id": &fakeUUID{uuid: "runid2"},
+						"workflow_id":    "wfid2",
+						"workflow_state": 1,
+					},
+				},
+				pageState: []byte("test-page-token-2"),
+			},
+			wantExecutions: []*persistence.CurrentWorkflowExecution{
+				{
+					DomainID:     "domain1",
+					WorkflowID:   "wfid1",
+					RunID:        "30000000-0000-f000-f000-000000000001",
+					State:        1,
+					CurrentRunID: "runid1",
+				},
+				{
+					DomainID:     "domain1",
+					WorkflowID:   "wfid2",
+					RunID:        "30000000-0000-f000-f000-000000000001",
+					State:        1,
+					CurrentRunID: "runid2",
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+
+			query := gocql.NewMockQuery(ctrl)
+			query.EXPECT().PageSize(tc.pageSize).Return(query).Times(1)
+			query.EXPECT().PageState(tc.pageToken).Return(query).Times(1)
+			query.EXPECT().WithContext(gomock.Any()).Return(query).Times(1)
+			if tc.iter != nil {
```

Comment:
nit:  I would consider setting up a per-test affordance for these if they're going to expand at all; lots of branching assertions can get pretty hard to read imho. 

ie
```
tests := []struct {
		name           string
		...
		querymock              func(m *mockQuery) // pass in what you expect the mock to return per row
	}{
...

```

Reply:
In general I agree but these are all shared mocks and each test case provides inputs to the function which are used to setup the mock here.